### PR TITLE
feat: async install

### PR DESCRIPTION
**Pull request by iamayushm**
Changes needed to make helm install and update async 
https://github.com/devtron-labs/devtron/issues/3482

### CODE REVIEW

File: pkg/util/k8s/K8sUtil.go [hunk 1]
```diff
@@ -86,10 +86,12 @@ func GetGvkVsChildGvrAndScope() map[schema.GroupVersionKind][]GvrAndScope {
...
-		restConfig, err = rest.InClusterConfig()
-		if err != nil {
-			return nil, err
-		}
+		//restConfig, err = rest.InClusterConfig()
+		//if err != nil {
+		//	return nil, err
+		//}
+		//return restConfig, err
+		restConfig = &rest.Config{Host: "https://20.119.53.88:16443", BearerToken: "SllJTi9qalZOQWIvWlgyOVUwcnlBRkxKSE9NeEJoa1Z5Ym5Ec1RPdWFXWT0K", TLSClientConfig: rest.TLSClientConfig{Insecure: true}}
```
Comment (vikramdevtron) [inline pkg/util/k8s/K8sUtil.go:94]: Why this hardcoded host

File: pkg/service/HelmAppServiceWrapper.go [hunk 2]
```diff
@@ -365,6 +368,7 @@ func (impl *ApplicationServiceServerImpl) AppDetailAdaptor(req *bean.AppDetail)
 			PodMetadata: podMetadatas,
 		},
 		EnvironmentDetails: req.EnvironmentDetails,
+		ReleaseExist:       true,
```
Comment (vikramdevtron) [inline pkg/service/HelmAppServiceWrapper.go:371]: IsReleaseExists

File: pkg/service/HelmAppService.go [hunk 3]
```diff
@@ -668,13 +687,26 @@ func (impl HelmAppServiceImpl) UpgradeReleaseWithChartInfo(ctx context.Context,
...
+		if err != nil {
+			impl.logger.Errorw("Error in update release ", "err", err)
+			helmInstallMessage := HelmInstallNatsMessage{
+				InstallAppVersionHistoryId: int(request.InstallAppVersionHistoryId),
+				Message:                    err.Error(),
+				IsReleaseInstalled:         false,
+			}
+			data, err := json.Marshal(helmInstallMessage)
+			if err != nil {
+				impl.logger.Errorw("error in marshalling nats message")
```
Comment (vikramdevtron) [inline pkg/service/HelmAppService.go:702]: missing return

File: pkg/service/HelmAppService.go [hunk 4]
```diff
@@ -584,13 +590,26 @@ func (impl HelmAppServiceImpl) installRelease(request *client.InstallReleaseRequ
...
+		if err != nil {
+			impl.logger.Errorw("Error in install release ", "err", err)
+			helmInstallMessage := HelmInstallNatsMessage{
+				InstallAppVersionHistoryId: int(request.InstallAppVersionHistoryId),
+				Message:                    err.Error(),
+				IsReleaseInstalled:         false,
+			}
+			data, err := json.Marshal(helmInstallMessage)
+			if err != nil {
+				impl.logger.Errorw("error in marshalling nats message")
```
Comment (vikramdevtron) [inline pkg/service/HelmAppService.go:604]: missing return statement

File: pkg/k8sInformer/K8sInformer.go [hunk 5]
```diff
@@ -428,6 +433,7 @@ func (impl *K8sInformerImpl) startInformerAndPopulateCache(clusterId int) error
 				impl.mutex.Lock()
 				defer impl.mutex.Unlock()
 				delete(impl.HelmListClusterMap[clusterId], releaseDTO.Name+string(rune(clusterModel.Id)))
+				delete(impl.HelmReleaseStatusMap[clusterId], releaseDTO.Name+string(rune(clusterModel.Id)))
```
Comment (vikramdevtron) [inline pkg/k8sInformer/K8sInformer.go:436]: concatenation could be better

File: pkg/k8sInformer/K8sInformer.go [hunk 6]
```diff
@@ -384,6 +387,7 @@ func (impl *K8sInformerImpl) startInformerAndPopulateCache(clusterId int) error
 				}
 				impl.mutex.Lock()
 				defer impl.mutex.Unlock()
+				impl.HelmReleaseStatusMap[clusterId][releaseDTO.Name+string(rune(clusterModel.Id))] = releaseDTO.Info
```
Comment (vikramdevtron) [inline pkg/k8sInformer/K8sInformer.go:390]: concatenation could be better

**Review by vikramdevtron — DISMISSED** [review]

File: pkg/service/HelmAppService.go (hunk 3 above)
Comment (iamayushm) [reply pkg/service/HelmAppService.go:862]: this is intentional

**Review by vikramdevtron — APPROVED** [review]

File: pkg/service/HelmAppService.go (hunk 4 above)
Comment (iamayushm) [reply pkg/service/HelmAppService.go:699]: this is intentional

**Pull request merged**